src/libostree/ostree-repo-checkout.c

```diff
@@ -362,6 +375,12 @@ checkout_file_hardlink (OstreeRepo                          *self,
     {
       ret_was_supported = FALSE;
     }
+  else if (errno == EEXIST && options->overwrite_mode == OSTREE_REPO_CHECKOUT_OVERWRITE_ADD_FILES)
+    {
+      /* In this mode, we keep existing content.  Still need to mark the hardlink as supported.
+       */
+      ret_was_supported = TRUE;
```

Comment:
Oooh.  That's a good catch.  That would have been spectacularly painful to debug later.